doc/source/whatsnew/v0.22.0.txt

```diff
@@ -139,6 +139,7 @@ Other Enhancements
 - :func:`read_excel()` has gained the ``nrows`` parameter (:issue:`16645`)
 - :func:``DataFrame.to_json`` and ``Series.to_json`` now accept an ``index`` argument which allows the user to exclude the index from the JSON output (:issue:`17394`)
 - ``IntervalIndex.to_tuples()`` has gained the ``na_tuple`` parameter to control whether NA is returned as a tuple of NA, or NA itself (:issue:`18756`)
+- :func:``DataFrame.assign()`` now acceepts dependent kwargs, e.g. `df.assign(b=1, c=lambda x:x['b'])` does not throw an exception anymore. (:issue: `14207)
```

Comment:
previously this would raise (put the what it did raise), write the issue like

``(:issue:`14207`)``

Reply:
Also:
- <code>:func:\`\`DataFrame.assign()\`\` </code> --> <code>:func:\`DataFrame.assign\`</code> (single ticks, no parenthesis)
- acceepts --> accepts
- <code>\`df.assign(b=1, c=lambda x:x['b'])\`</code> --> <code>\`\`df.assign(b=1, c=lambda x: x['b'])\`\`</code> (double ticks)

Reply:
I think it'd be worth breaking this out into its own section and highlight a case where this is an API change. Something like


    .. warning::

    This may subtly change the behavior of your code when you're
    using ``assign`` to update an existing column. Previously, callables
    refering to other variables being updated would get the "old" values

    .. code-block:: ipython

        In [2]: df = pd.DataFrame({"A": [1, 2, 3]})

        In [3]: df.assign(A=lambda df: df.A + 1, C=lambda df: df.A * -1)
        Out[3]:
            A  C
        0  2 -1
        1  3 -2
        2  4 -3

    Now, callables will get the "new" value

    .. ipython:: python

        df = pd.DataFrame({"A": [1, 2, 3]})
        df.assign(A=lambda df: df.A + 1, C=lambda df: df.A * -1)